engine/cdn/artifacts.go

```diff
@@ -0,0 +1,60 @@
+package cdn
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ovh/cds/sdk"
+)
+
+func (s *Service) storeArtifact(ctx context.Context, body io.ReadCloser, cdnRequest sdk.CDNRequest) (*sdk.WorkflowNodeRunArtifact, error) {
+	art := cdnRequest.Artifact
+	if _, err := art.IsValid(); err != nil {
+		return nil, sdk.WrapError(err, "artifact is not valid")
+	}
+	storageDriver, err := s.getDriver(cdnRequest.ProjectKey, cdnRequest.IntegrationName)
```

Comment:
Why not push it in first layer cache , and then upload in storage + mirror after ?   to be fast